pkg/kvstore/kvstore.go

```diff
@@ -32,54 +34,59 @@ var (
 // This package implements a very basic key/value store.  It is up to the
 // caller to provision the namespace.
 
-type KeyValueStore struct {
-	b Backend
-
-	kv map[string][]byte
-
-	fileName string
-
-	l sync.RWMutex
+type KeyValueStore interface {
+	Set(ctx context.Context, key string, value []byte) error
+	Get(key string) ([]byte, error)
+	List(re string) (map[string][]byte, error)
+	Delete(ctx context.Context, key string) error
+	Save(ctx context.Context) error
+	Name() string
 }
 
-type Backend interface {
-	// Creates path and overwrites whatever is there.
-	Upload(ctx context.Context, r io.Reader, pth string) error
-
-	// Downloads from the given path.
-	Download(ctx context.Context, pth string) (io.ReadCloser, error)
+type kv struct {
+	b    Backend
+	kv   map[string][]byte
+	name string
+	l    sync.RWMutex
+}
 
-	// Moves the given path.
-	Mv(ctx context.Context, fromPath, toPath string) error
+func fileName(name string) string {
+	return fmt.Sprintf("%s.dat", name)
 }
 
 // Create a new KeyValueStore instance using the given Backend with the given
 // file.  If the file exists on the Backend, it is restored.
-func NewKeyValueStore(op trace.Operation, store Backend, fileName string) (*KeyValueStore, error) {
-	p := &KeyValueStore{
-		b:        store,
-		kv:       make(map[string][]byte),
-		fileName: fileName,
+func NewKeyValueStore(ctx context.Context, store Backend, name string) (KeyValueStore, error) {
```

Comment:
I think its better to not use it here, and use it in higher level calls instead. It makes this API both compatible with context.Context and trace.Operation (since a trace.Operation can be passed in).